roles/common/tasks/main.yml

```diff
@@ -10,7 +16,7 @@
    - unzip
    - which
    - java-1.7.0-openjdk-devel
-  when: ansible_pkg_mgr == 'yum'
+  when: ansible_pkg_mgr == 'yum' and (java_version | failed or java_version.stdout | version_compare('1.7', '<'))
```

Comment:
I'm still an ansible noob here, but wouldn't this NOT install unzip and which if java version is already 1.7 and above?